src/modules/storage/LedgerStorage.ts

```diff
@@ -1318,4 +1318,42 @@ export class LedgerStorage extends Storages
 
         return this.query(sql, [address]);
     }
+
+    /**
+     * Gets a transaction data
+     * @param tx_hash The hash of the transaction to get
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called with the records
+     * and if an error occurs the `.catch` is called with an error.
+     */
+    public getTransactionPending (tx_hash: Hash): Promise<Transaction | null>
+    {
+        return new Promise<Transaction | null>(async (resolve, reject) =>
+        {
+            try {
+                let hash = tx_hash.toBinary(Endian.Little);
+                let rows = await this.query("SELECT * FROM transaction_pool WHERE tx_hash = ?;", [hash]);
+                if (rows.length > 0)
+                {
+                    let input_rows = await this.query("SELECT * FROM tx_input_pool WHERE tx_hash = ? ORDER BY input_index;", [hash]);
+                    let output_rows = await this.query("SELECT * FROM tx_output_pool WHERE tx_hash = ? ORDER BY output_index;", [hash]);
+
+                    let inputs:Array<TxInput> = [];
+                    for (let input_row of input_rows)
+                        inputs.push(new TxInput(new Hash(input_row.utxo, Endian.Little), new Signature(input_row.signature, Endian.Little)));
+                    let outputs:Array<TxOutput> = [];
+                    for (let output_row of output_rows)
+                        outputs.push(new TxOutput(output_row.amount, new PublicKey(output_row.address)));
+                    resolve(new Transaction(rows[0].type, inputs, outputs, new DataPayload(rows[0].payload, Endian.Little)));
```

Comment:
How about moving the data-type conversion capabilities from LedgerStorage to Stoa?